### ISSUE TITLE
Support for Newer TPLs Via Configure

### ISSUE THREAD

**Issue by rspavel** [HPCToolkit/hpctoolkit#111]
I'm currently working through updating the spackage for hpctoolkit and am running in to issues with things like the libiberty path being hardcoded and conflicting with wether binutils chooses lib or lib64 for a library. While this has obviously only been tested with specific library versions, many of these conflicts can likely be resolved with the patched versions (or very careful specification of version numbers during concretization).
And this will be useful on platforms where we may need those updated libraries due to toolchain issues.

Would it be possible to expose all of the dependencies at the hpctoolkit (or hpctoolkit-external) level for the purpose of allowing users to specify their own installations for every external?

Most (all?) of the externals seem to be open source projects and most are actually in spack itself. I already updated the externals spackage to use spack available versions where possible, but I don't see an option to specify paths to some of the libraries (specifically dyninst/symtabAPI, boost, and lzma, but optionally everything).

**Comment by laksono** [comment]
Hi Robert,
We don't get notification from github if someone submit a new issue. Please send your bug to hpctoolkit-forum@rice.edu
Concerning spack package and hpctoolkit-externals, @mwkrentel leads the effort to port hpctoolkit and hpctoolkit-externals to spack.

**Comment by laksono** [comment]
@jmellorcrummey, @mwkrentel: how llnl can set github to notify if there is a new issue ?

**Comment by mwkrentel** [comment]
What is TPL?

And how are you updating spackage for hpctoolkit?
You're editing the spack hpctoolkit package.py recipe?

If I understand correctly, then I'm already doing exactly that.  I'm
rewriting the spack hpctoolkit package.py file to specify 15 or so
dependencies and take them directly from spack.  I have something that
works for "simple" cases, ie not Cray and not Blue Gene.

I'm waiting on two things.  (1) Dyninst will soon release 10.x, and
(2) I have a branch of hpctoolkit that I'll merge into master that
will be more spack-friendly.  When those are in place, then I'll
replace package.py for hpctoolkit and delete hpctoolkit-externals.

As for versions, my plan is to have minimal requirements in hpctoolkit
package.py and specify versions with packages.yaml.  We'll maintain a
reference version, but yes, it would be easy for someone to swap in a
different version (along with all the headaches of everyone running a
different configuration).

Maybe you want to let me handle this?

You're at LLNL?   I've been working with @scheibelp on this.

**Comment by rspavel** [comment]
I am not at LLNL. I was just trying to fix the hpctoolkit spackage to make it easier to use as we are investigating this as an addition to our workflows.
I have externals building but with the current concretization issues and hpctoolkit itself one or two fixes away from functionality. But if that is already under control then we'll just hold off on using this until it works better with our infrastructure.

Thanks

**Comment by mwkrentel** [comment]
The thing about the existing spack packages for toolkit and externals
is that we didn't write them.

I'm working on replacing the hpctoolkit recipe.  I've made a lot of
changes to hpctoolkit and edited 9 spack recipes to be compatible.
But this depends on a branch that I'm not quite ready to merge just
yet.

Until then, the officially supported build is outside of spack with
'configure, make, make install' for externals and hpctoolkit.

Does something in the spack package not build for you?

I actually haven't tried the spack hpctoolkit recipe.  But looking at
the package.py files, I expect that they would work for "simple"
cases.  There are no variants for some necessary adjustments for Cray
or Blue Gene, so they won't work there.  But for a generic build, I
expect they should work.  Use version 'master' for both.

> I am not at LLNL.

Sorry, I misread your github page.  Every time I see LANL my brain
stores this as LLNL.  Every time.  Sorry about that.  :-(

**Comment by mwkrentel** [comment]
@rspavel I've rewritten the spack hpctoolkit package.py file to build
add the dependencies from spack instead of hpctoolkit-externals and
pass all of the dependencies via --with-pkg.  I believe that is what
you were asking for.

I have directions for building from spack and a 'reference'
packages.yaml file in the spack subdirectory in the hpctoolkit repo.

```
-rw-r--r-- 1 krentel hpc 20885 Jan 29 14:22 README.spack
-rw-r--r-- 1 krentel hpc 10716 Jan 29 14:21 packages.yaml
```

There are two things I'm waiting on and then I'll announce 'done' on
the hpctoolkit-forum mailing list.

1. Building on Cray requires spack PR #10369.  I've started nagging
people about this, so probably it will be merged soon.

2. Dyninst currently breaks with boost 1.69.  I've submitted spack PR
#10454 to fix this.  In the meantime, use boost <= 1.68.

Is there anything else, or are we ready to close?
